docs/sources/fundamentals/architecture/rings.md

```diff
@@ -4,4 +4,87 @@ weight: 40
 ---
 # Consistent Hash Rings
 
-TODO: describe why and how consistent hash rings are used.
+[Consistent hash rings](https://en.wikipedia.org/wiki/Consistent_hashing)
+are incorporated into Loki cluster architectures to
+
+- aid in the sharding of log lines
+- implement high availability 
+- ease the horizontal scale up and scale down of clusters.
+There is less of a performance hit for operations that must rebalance data.
+
+Hash rings connect instances of a single type of component when 
+
+- there are a set of Loki clusters in monolithic deployment mode
+- there are multiple read components or multiple write components in
+simple scalable deployment mode
+- there are multiple instances of one type of component in microservices mode
+
+Not all Loki components are connected by hash rings.
+These components need to be connected into a hash ring:
+
+- distributors
+- ingesters
+- query schedulers
+- compactors
+- rulers
+
+In an architecture that has three distributors and three ingestors defined,
+the hash rings for these components connect the instances of same-type components.
+
+![distributor and ingester rings](../ring-overview.png)
+
+Each node in the ring represents an instance of a component.
+Each node has a key-value store that holds communication information
+for each of the nodes in that ring.
+A [gossip protocol](https://en.wikipedia.org/wiki/Gossip_protocol) propagates the information to all the nodes
+to guarantee the eventual consistency of the key-value store contents.
```

Comment:
This looks out of the context because gossip(memberlist) is one of the supported stores for the ring. You can maybe say each node periodically keeps updating its info in the store at the configured heartbeat interval.

Reply:
I don't understand your comment.  To me, there are 2 different issues.  Issue 1: which store is used. Issue 2: the gossip protocol is used to propagate info.  

gossip(memberlist) is a particular implementation detail and not about how the rings work (in general).

Are you telling me that a different algorithm is used to propagate info if a store type other than memberlist is used? If yes, then I will change the prose.  

My prose is there to explain to readers the basics of how the ring works.  I'm trying to do this independent of which configuration/implementation they choose for their key-value store.

Reply:
Yes, I think not all available stores use the gossip protocol (ex: etcd and inmemory).